File: code/scenes/src/main/java/org/fbme/scenes/controllers/components/ComponentExtController.java
package org.fbme.scenes.controllers.components;

import jetbrains.mps.nodeEditor.cells.EditorCell;
import org.jetbrains.annotations.NotNull;

import java.awt.*;

public interface ComponentExtController<FormT> {

    @NotNull
    EditorCell getComponentCell();

    void updateCellWithForm(FormT form);

    void updateCellSelection(boolean selected);

    void paintTrace(Graphics g, FormT form);
}

File: code/scenes/src/main/kotlin/org/fbme/scenes/controllers/components/ComponentExtController.kt
package org.fbme.scenes.controllers.components

import jetbrains.mps.nodeEditor.cells.EditorCell
import java.awt.Graphics
import java.awt.Point

interface ComponentExtController<FormT> {
    val componentCell: EditorCell
    fun getCoordinates(form: FormT): Point
    fun updateCellWithForm(form: FormT)
    fun updateCellSelection(selected: Boolean)
    fun paintTrace(g: Graphics, form: FormT)
}
